api/csi.go

```diff
@@ -79,19 +83,25 @@ const (
 
 // CSIVolume is used for serialization, see also nomad/structs/csi.go
 type CSIVolume struct {
-	ID             string
-	Namespace      string
-	Name           string
-	ExternalID     string
-	Topologies     []*CSITopology
-	AccessMode     CSIVolumeAccessMode
-	AttachmentMode CSIVolumeAttachmentMode
-
-	// Combine structs.{Read,Write,Past}Allocs
+	ID             string                  `hcl:"id"`
+	Name           string                  `hcl:"name"`
+	ExternalID     string                  `hcl:"external_id"`
+	Namespace      string                  `hcl:"namespace"`
+	Topologies     []*CSITopology          `hcl:"topologies"`
+	AccessMode     CSIVolumeAccessMode     `hcl:"access_mode"`
+	AttachmentMode CSIVolumeAttachmentMode `hcl:"attachment_mode"`
+
+	// Allocations, tracking claim status
+	ReadAllocs  map[string]*Allocation
+	WriteAllocs map[string]*Allocation
+
+	// Combine structs.{Read,Write}Allocs
 	Allocations []*AllocationListStub
 
+	// Healthy is true iff all the denormalized plugin health fields are true, and the
```

Comment:
We changed the name of the field, so the comment should match. We also only mark Schedulable based on plugin health and not for marking the volume for GC:

```go
	vol.Schedulable = vol.NodesHealthy > 0
	if vol.ControllerRequired {
		vol.Schedulable = vol.ControllersHealthy > 0 && vol.Schedulable
	}
```

Reply:
ugh, stale comment! 👍